routes/index.ts

```
@@ -266,19 +267,24 @@ const getCatch = async (req, res) => {
 
 const newCatch = async (req, res) => {
     if (req.headers['content-type'] == "application/xml") { stringParser(req); }
-    setTimeout( () => {
+    setTimeout(async () => {
         if (req.params.tripNum && req.body.tripNum && req.body.source && req.body.hauls) {
-                const newTrip = req.body;
-                newTrip.type = 'trips-api-catch';
-                newTrip.createdDate = moment().format();
-                masterDev.bulk({docs: [newTrip]}).then(
+            const newTrip = req.body;
+            newTrip.type = 'trips-api-catch';
+            newTrip.createdDate = moment().format();
+
+            const doc = await masterDev.view('TripsApi', 'all_api_catch', { "key": req.params.tripNum });
```

This isn't quite the check we need to perform.  API tripCatch submissions may be the first trips-api-catch doc for a tripNum, and there may be multiple trips-api-catch docs for a tripNum...

We want to validate that a trip with the supplied tripNum exists by querying:
masterDev.view('TripsApi', 'all_api_trips', { "key": req.params.tripNum });  // look for trip rather than catch
If no rows are returned, the API should throw an error.  

If a row is returned (there is a valid trip), the api should write the new trips-api-catch unless there is already a trips-api-catch with both the same tripNum and the same source.  (we would expect to get both a source: logbook and a source: thirdParty with the same tripNum for many trips.

---

okay added checks to query tripAPI to see if tripNum exists. And only writes new doc if tripNum exists and source is not the same